metsim/methods/mtclim.py

```diff
@@ -89,72 +88,72 @@ def _calc_tfmax(precip, dtr, sm_dtr):
         warn('missing values with unsmoothed data')
         sm_dtr.fillna(dtr, inplace=True)
 
-    # Calculate annual total precip
-    sum_precip = df['precip'].values.sum()
-    ann_precip = (sum_precip / params['n_days']) * cnst.DAYS_PER_YEAR
-    if ann_precip == 0.0:
-        ann_precip = 1.0
+    # Calculate annual total prec
+    sum_prec = df['prec'].values.sum()
+    ann_prec = (sum_prec / params['n_days']) * cnst.DAYS_PER_YEAR
+    if ann_prec == 0.0:
+        ann_prec = 1.0
 
-    # Effective annual precip
+    # Effective annual prec
     if params['n_days'] <= 90:
-        # Simple scaled method, minimum of 8 cm 
-        sum_precip = df['precip'].values.sum()
-        eff_ann_precip = (sum_precip / params['n_days']) * cnst.DAYS_PER_YEAR
-        eff_ann_precip = np.maximum(eff_ann_precip, 8.0)
-        parray = eff_ann_precip
+        # Simple scaled method, minimum of 8 cm
+        sum_prec = df['prec'].values.sum()
+        eff_ann_prec = (sum_prec / params['n_days']) * cnst.DAYS_PER_YEAR
+        eff_ann_prec = np.maximum(eff_ann_prec, 8.0)
+        parray = pd.Series(eff_ann_prec, index=df.index)
     else:
-        # Calculate effective annual precip using 3 month moving window
+        # Calculate effective annual prec using 3 month moving window
         window = pd.Series(np.zeros(params['n_days'] + 90))
-        window[90:] = df['precip']
-        
+        window[90:] = df['prec']
+
         # If yeardays at end match with those at beginning we can use
         # the end of the input to generate the beginning by looping around
         # If not, just duplicate the first 90 days
-        start_day, end_day = df['day_of_year'][0], df['day_of_year'][-1]
+        start_day, end_day = df.index.dayofyear[0], df.index.dayofyear[-1]
         if (start_day%365 == (end_day%365)+1) or (start_day%366 == (end_day%366)+1):
-            window[:90] = df['precip'][-90:]
+            window[:90] = df['prec'][-90:]
         else:
-            window[:90] = df['precip'][:90]
+            window[:90] = df['prec'][:90]
 
-        parray = (window.rolling(window=90, win_type=win_type, axis=0) 
+        parray = (window.rolling(window=90, win_type=win_type, axis=0)
                     .mean()[90:] * cnst.DAYS_PER_YEAR)
 
-    # Convert to mm 
+    # Convert to cm
     parray = parray.where(parray>80.0, 80.0) / cnst.MM_PER_CM
```

Comment:
80mm was the minimum annual precipitation set in the VIC code, to avoid very large PET to annual precip ratios.  I don't know why 80 was specifically chosen.